pkg/operator/controller/ingress/controller.go

```diff
@@ -134,9 +134,51 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(source.Kind(operatorCache, &configv1.Proxy{}), handler.EnqueueRequestsFromMapFunc(reconciler.ingressConfigToIngressController)); err != nil {
 		return nil, err
 	}
+	// Watch secrets in the operand namespace that are referenced as default certificates so that
+	// checkDefaultCertificate can determine whether changes to the default certificate affect upgradeability.
+	// This watch is NOT required for updating the router deployment, as its volume mount automatically
+	// handles updates to the default certificate.
+	if err := c.Watch(source.Kind(operatorCache, &corev1.Secret{}), handler.EnqueueRequestsFromMapFunc(reconciler.defaultCertificateSecretToIngressController), predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Object.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.(*corev1.Secret).Namespace == operatorcontroller.DefaultOperandNamespace
+		},
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}); err != nil {
+		return nil, err
+	}
 	return c, nil
 }
 
+// defaultCertificateSecretToIngressController creates reconcile requests for
+// IngressControllers that reference the provided secret as their default certificate.
+func (r *reconciler) defaultCertificateSecretToIngressController(ctx context.Context, o client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+	secret := o.(*corev1.Secret)
+	controllers := &operatorv1.IngressControllerList{}
+	if err := r.cache.List(ctx, controllers, client.InNamespace(r.config.Namespace)); err != nil {
+		log.Error(err, "failed to list ingresscontrollers for default certificate secret", "related", o.GetSelfLink())
+		return requests
+	}
+	for _, ic := range controllers.Items {
+		if ic.Spec.DefaultCertificate != nil && ic.Spec.DefaultCertificate.Name == secret.Name {
```

Comment:
`ic.Spec.DefaultCertificate == nil` implies that it's an operator-generated secret, and in the context of the upgradeability check, I suppose we don't really need to check the secret if it's an operator-generated secret.  However, `defaultCertificateSecretToIngressController` itself isn't obviously applicable only to custom default certificates.  Could you either add a comment to make this explicit, or make sure to cover operator-generated secrets?  For the latter option, I think using `RouterEffectiveDefaultCertificateSecretName` would be sufficient and simple.

Reply:
Ah good point, didn't think about the operator generated secret.

Since `checkDefaultCertificate` also checks the operator generated secret, it makes sense to reconcile changes to it to be complete.